test/optim/test_optim.py

```diff
@@ -48,6 +54,75 @@ def _diff_fn(p, grad, opt_differentiable_state, opt_class, kwargs, *ignored):
     )
 
 
+def _multistep_backprop_diff_lr_fn(
+    params: Tensor,
+    grad: Tensor,
+    inner_opt_differentiable_state: dict[str, Any],
+    inner_opt_class: type[Optimizer],
+    inner_kwargs: dict[str, Any],
+    *ignored: Any,
+) -> tuple[Tensor, ...]:
+    assert (
+        inner_kwargs["differentiable"] is True
+    ), "Only call this test function when differentiable=True"
+    # TODO: Adjust this function to test more hyperparameters than just lr
+    assert "lr" in inner_kwargs, "Only call this test function with a custom lr"
```

Comment:
Same here

Reply:
This one was similar because I wanted to expand this later to be "assert 'lr' in inner_kwargs or 'betas' in inner_kwargs' or 'weight_decay' in inner_kwargs' "